modules/cms/classes/Controller.php

```diff
@@ -148,18 +148,16 @@ public function run($url = '/')
             $url = Request::path();
         }
 
-        if (!strlen($url)) {
+        if ('' === $url) {
```

Comment:
Who invited yoda? 😄 In all seriousness though, wouldn't an `if (empty())` be better here?

Reply:
=) funny, I would rather switch sides (totally agree with stick to common style), then make this statements with empty as `empty()` checks for many (`null`, `false` etc) values, which is expensive, only in those cases where we expecting `string` as input value, like in this example.

Reply:
Are you sure it's that expensive? Run some benchmarks on it and then get back to me with the results. `empty()` is a language construct, it's pretty fast. I personally prefer using it because it's easier to read for developers and it encompasses more potential states for the variable to be in that all ultimately mean the same thing, that there is no useful data in the variable to work with.

Reply:
Definitely, the decision is yours - tests:
```php
<?php
// empty
$start = microtime(true);
$str = '';
for ($i = 0;$i < 1e6;++$i) {
    if (empty($str)) {

    }
}
$t = microtime(true) - $start;
echo 'empty time:' . $t . PHP_EOL;
// equation
$start = microtime(true);
for ($i = 0;$i < 1e6;++$i) {
    if ($str === '') {

    }
}
$t = microtime(true) - $start;
echo 'string time:' . $t . PHP_EOL;
```  

Results:
1...
```
empty time:0.14257001876831
string time:0.1290819644928
```
2...
```
empty time:0.14147520065308
string time:0.12645101547241
```
3...
```
empty time:0.1415810585022
string time:0.12579417228699
```

Reply:
Looks like approximately a hundredth of a second's difference over 1 million iterations in favour of `"" ===` over `empty()`. Personally in my mind that's not really enough to make `empty()` less desirable. Do you have any further input on it though? (Also, `$str` is defined in the empty test and reused in the `=== ''` test, so that might make a very minor difference).

Reply:
Ok, I'll turn all the "" === $str into empty()